### DISCUSSION TITLE
Deployment on Debian 12

### DISCUSSION THREAD

**Discussion by bambosd** [Lissy93/dashy#1753] [Help and Support]
Hello everyone. 
I'm excited to be involved and use dashy.  I'm searching for a guidance on how to deploy the web app on a debian system. 

I have found so far different and several methods suggesting installing the dependencies and the git clone the repository. 
Then i got a different story each time for compile / install / run leading to all sorts of errors. my latest was with npm start : error initializing - check logs.  I could use some enlightenment here.

**Comment by mycoolh** [comment]
1. **Install Node.js and npm**:
   Run these commands to install Node.js and npm, which are required to run Dashy:

   ```bash
   sudo apt update
   sudo apt install nodejs npm
   ```

2. **Clone the Dashy Repository**:
   Use Git to clone the Dashy project:

   ```bash
   git clone https://github.com/Lissy93/dashy.git
   cd dashy
   ```

3. **Install Dependencies**:
   Inside the Dashy directory, install the required packages:

   ```bash
   npm install
   ```

4. **Start Dashy**:
   Run Dashy with the following command:

   ```bash
   npm start
   ```

5. **Access Dashy**:
   Open your web browser and go to `http://localhost:8080` to access the Dashy interface.

### Troubleshooting

- **If `npm start` gives an error**:
  - Check the terminal output for specific error messages.
  - Run `npm install` again to ensure all dependencies are installed correctly.
  - Make sure your network and firewall settings allow traffic on port 8080.

**Reply by bambosd** [reply]
Hello Sir and thank you for the prompt reply.

during npm install several warning appear stating unsupported engines , even if i install the dependencies on debian 12.7
![image](https://github.com/user-attachments/assets/0cb2c132-0cd4-44ee-8b9b-b91454d10a6e)
At the end of npm install : the error is :
Some issues need review, and may require choosing a different dependency.

The npm start commands is running the web server on port 4000, that is accessible on my networks, and the web server seems to load.
After the initial splash screens, i got the following error:
![image](https://github.com/user-attachments/assets/777253f9-6a09-4bb4-b881-1377e89f2a7c)

thanks for any suggestions.

**Reply by bambosd** [reply]
@Utopiality7 any comments on that ?? Is there any other route to do that ?
